hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/GlobalDeleteKeyGenerator.java

```diff
@@ -60,13 +60,13 @@ public String getPartitionPath(GenericRecord record) {
   @Override
   public String getRecordKey(Row row) {
     tryInitRowAccessor(row.schema());
-    return combineCompositeRecordKey(rowAccessor.getRecordKeyParts(row));
+    return combineCompositeRecordKey(true, rowAccessor.getRecordKeyParts(row));
```

Comment:
why the flag doesn't follow the write config `COMPLEX_KEYGEN_ENCODE_SINGLE_RECORD_KEY_FIELD_NAME` ?

Reply:
`NonpartitionedAvroKeyGenerator` also needs a fix.